src/astgen.zig

```diff
@@ -1762,8 +1762,9 @@ const CondKind = union(enum) {
 
     fn thenSubScope(self: CondKind, mod: *Module, then_scope: *Scope.GenZIR, src: usize, payload_node: ?*ast.Node) !*Scope {
         if (self == .bool) return &then_scope.base;
-
-        const payload = payload_node.?.castTag(.PointerPayload) orelse {
+        const payload = if (payload_node) |node|
```

Comment:
We return early if `self == .bool` so `payload_node` must be non-null.

Reply:
Oh huh, it triggered the .? before but now it doesn't. Thanks.